### PR TITLE
Map item processing refactoring

### PR DESCRIPTION
**Pull request by Hirmuolio**
#### Summary

```SUMMARY: Infrastructure "Map item processing refactoring"```

#### Purpose of change

The chain of functions for processing items on map is weird and hard to follow.
There are also many things that are in there but do not do anything.

#### Describe the solution

* The processing functions pases along `processor` pointer to a function. The function is always the same `process_map_items` so this was pointless. Now they just call the function.
* There is a string signal passed around. It is not used anywhere for anything. Removed `signal`.
* z position is passed along in vehicle item processing but not used for anything. Removed `gridz`.
* Boolean in `process_items` is always true and was only used in an or statement where it did nothing. Removed the boolean.
* The only thing that `process_active_items` did was set the above unneeded boolean to true and then call `process_items`. Removed `process_active_items` and instead just call `process_items`. `process_items` was made public so it can be called.
* `process_item` always passes along a boolean false. Just have false in the final palce.
* `process_map_items` does nothing but call `process_item`. Replace `process_item` with `process_map_items`.

#### Describe alternatives you've considered

Let it stay as unreadable mess of functions calling other functions calling other functions without clearly stating what function is called while passing variables that do nothing.

#### Testing

Tests pass, items still get processed in inventory, on ground and in vehicles.

#### Additional context

There should be zero visible changes in gameplay.

### CODE REVIEW

**Comment by KorGgenT** [comment]
probably conflicts with #39406

**Comment by Hirmuolio** [comment]
You haven't made any edits in these areas at least yet.

**Comment by KorGgenT** [comment]
oh good, i was worried it would cause a merge conflict like the last one. i only skimmed it, i didn't try to review it.
